### ISSUE TITLE
previewData() does not work with ./app folder structure on Next.js 3.3.x

### ISSUE THREAD

**Issue by daniel-souza** [vercel/next.js#48730]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
`yarn dev` -> `next dev` or `next build` -> `next start`
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

https://gitpod.io#snapshot/4079a73c-e8ea-4175-9634-cf2d5873bfec

### To Reproduce

Basically it's required two routes:
- `api/preview` - that enables the preview mode
```typescript
import { NextApiRequest, NextApiResponse } from "next";

export default function preview(req: NextApiRequest, res: NextApiResponse) {
    res.setPreviewData({});
    res.writeHead(307, {Location: '/'});
    res.end();
    // res.end("Preview mode enabled");
    // res.redirect("/");
}
```
- `api/exit-preview` - that clear the cookies related to preview mode.
```typescript
import { NextApiRequest, NextApiResponse } from "next";

export default function exit(req: NextApiRequest, res: NextApiResponse) {
    res.clearPreviewData({});
    res.writeHead(307, { Location: "/" });
    res.end();
}
```

Next a simple page mapped to `./app/(user)/page.tsx`

```typescript
import { previewData } from "next/headers"

function HomePage() {
    console.log(previewData())
    if(previewData()) {
        return <div>Preview Mode</div>
    }
    return <div>Not Preview Mode</div>
}
```
 
The behaviour I got was:
- I access the route `/api/preview`
- the cookies __prerender_bypass and __next_preview_data are set.
- after redirection to `/` (the `./app/(user)/page.tsx`) the cookies are cleared.
- When loading the page, the log output from `previewData()` are evaluated always to `false`
- Finally, it only renders `Not Preview Mode`/

### Describe the Bug

The bug is: `previewData()` always evaluates to false, even enabling the preview mode using the `/api/preview` route.

With 'getStaticProps' I could access the `context.preview` and `context.previewData`.

However `"getStaticProps" is not supported in app/. Read more: https://beta.nextjs.org/docs/data-fetching/fundamentals`

in app dir, we can only use `getInitialProps`, but it does not have in its context the props of `preview` or `previewData`.

Its a bug because it works on version 3.2.x

### Expected Behavior

When acessing the route:

- `/api/preview`: the variable `previewData()` on `./app/(user)/page.tsx` must not be false and the rendered page must present the div with "Preview Mode" text.
- `/api/exit-preview`: the variable `previewData()` on `./app/(user)/page.tsx` must be false and the rendered page must present the div with "Not Preview Mode" text.

### How are you deploying your application? (if relevant)

yarn dev or yarn build then yarn start

**Comment by som-R91** [comment]
I'm also facing the exact same issue right now. This is my first project on Next.js, so I was stuck thinking I set this up incorrectly. Thanks for pointing out this bug.

**Comment by sebastianhaiss** [comment]
Same issue for me as well. Checking for `previewData()?.token` now always returns `false`, while previously it would return the expected token. 😮‍💨

**Comment by TheIsrael1** [comment]
God damn

**Comment by akashKarmakar02** [comment]
I'm also facing the same bug can you provide a temporary solution
I'm new to nextJs and can't find the solution

**Comment by kb1995** [comment]
I came across this issue while following this [repo](https://github.com/jpkha/nextjs13-storyblok) - even downgrading to 13.2.0 didn't fix the issue for me...

**Comment by AeroFiTech** [comment]
Yes facing the same issue with res.setPreviewData() on NextJS 13.3 

Code @ app/api/preview.ts
```javascript
import { NextApiRequest, NextApiResponse } from "next"

export async function GET(req: NextApiRequest, res: NextApiResponse) {
    res.setPreviewData({})
    res.writeHead(307, {Location: '/'})
    res.end()
}
``` 

Thanks for opening the issue to the awesome team!

**Comment by odinugel** [comment]
Having the same issue (v13.3.4)
On redirect to '/', the server deletes the cookies (expiration). This is not the case in v13.1.6

Request

| Name                   | Value                                                         | Domain     | Path | Expiration                  | Max-Age | Secure | SameSite | Priority |
|-----------------------|---------------------------------------------------------------|------------|------|-----------------------------|---------|--------|----------|----------|
| __prerender_bypass    | 814e83a...                               | localhost  | /    | Session                     | 50      | ✓      | Lax      | Medium   |
| __next_preview_data   | eyJhbGc9... | localhost  | /    | Session                     | 399     | ✓      | Lax      | Medium   |

Response
| Name                   | Value                                                         | Domain     | Path | Expiration                  | Max-Age | Secure |SameSite | Priority |
|-----------------------|---------------------------------------------------------------|------------|------|-----------------------------|---------|--------|----------|----------|
| __prerender_bypass    |                                                               | localhost  | /    | 1970-01-01T00:00:00.000Z     | 91      | ✓      | Lax      | Medium   |
| __next_preview_data   |                                                               | localhost  | /    | 1970-01-01T00:00:00.000Z     | 91      | ✓      | Lax      | Medium   |

**Comment by odinugel** [comment]
Updating to 13.4.0 and using the new [draftMode()](https://nextjs.org/docs/app/building-your-application/configuring/draft-mode) solved the issue. previewMode is considered legacy according to the new [docs](https://nextjs.org/docs/pages/building-your-application/configuring/preview-mode).
